extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/internal/DmxBridgeHandler.java

```diff
@@ -0,0 +1,214 @@
+        universe.unregisterChannels(thing);
+    }
+
+    /**
+     * get the universe associated with this bridge
+     *
+     * @return
+     *         the DMX universe id
+     */
+    public int getUniverseId() {
+        return universe.getUniverseId();
+    }
+
+    /**
+     * rename the universe associated with this bridge
+     *
+     * @param universeId
+     *            the new DMX universe id
+     */
+    protected void renameUniverse(int universeId) {
+        universe.rename(universeId);
+    }
+
+    @Override
+    public void thingUpdated(Thing thing) {
+        updateConfiguration();
+    }
+
+    /**
+     * open the connection to send DMX data to
+     */
+    protected abstract void openConnection();
+
+    /**
+     * close the connection to send DMX data to
+     */
+    protected abstract void closeConnection();
+
+    /**
+     * send the buffer of the current universe
+     */
+    protected abstract void sendDmxData();
+
+    /**
+     * install the sending and updating scheduler
+     */
+    protected void installScheduler() {
+        if (senderJob != null) {
+            uninstallScheduler();
+        }
+        if (refreshTime > 0) {
+            senderJob = scheduler.scheduleAtFixedRate(packetSender, 1, refreshTime, TimeUnit.MILLISECONDS);
+            logger.trace("started scheduler for thing {}", this.thing.getUID());
+        } else {
+            logger.info("refresh disabled for thing {}", this.thing.getUID());
+        }
+    }
+
+    /**
+     * uninstall the sending and updating scheduler
+     */
+    protected void uninstallScheduler() {
+        if (senderJob != null) {
+            if (!senderJob.isCancelled()) {
+                senderJob.cancel(true);
+            }
+            senderJob = null;
+            closeConnection();
+            logger.trace("stopping scheduler for thing {}", this.thing.getUID());
+        }
+    }
+
+    @Override
+    public void childHandlerDisposed(ThingHandler thingHandler, Thing thing) {
+        universe.unregisterChannels(thing);
+    }
+
+    /**
+     * get the configuration and update the bridge
+     */
+
+    protected void updateConfiguration() {
+        Configuration configuration = getConfig();
+
+        if (configuration.get(CONFIG_APPLY_CURVE) != null) {
+            universe.setDimCurveChannels((String) configuration.get(CONFIG_APPLY_CURVE));
+        }
+        if (configuration.get(CONFIG_REFRESH_RATE) != null) {
+            float refreshRate = ((BigDecimal) configuration.get(CONFIG_REFRESH_RATE)).floatValue();
+            if (refreshRate > 0) {
+                refreshTime = (int) (1000.0 / refreshRate);
+            } else {
+                refreshTime = 0;
+            }
+        } else {
+            refreshTime = 1000 / DEFAULT_REFRESH_RATE;
+        }
+        logger.debug("set refreshTime to {} ms in thing {}", refreshTime, this.thing.getUID());
+
+        installScheduler();
+    }
+
+    @Override
+    public void initialize() {
```

Comment:
Interesting. Removing it works now. Eclipse complained before if I didn't include it. Maybe that was before I changed it to "abstract".